### PR TITLE
Adding section for Optimization API

### LINKED ISSUE #358: Add Optimization API section
Optimization API needs its own page and should be a sidebar item within https://www.mapbox.com/android-docs/mapbox-services/overview/

Optimization example in demo app for reference: https://github.com/mapbox/mapbox-android-demo/blob/0a39320f3acd450eb45d6cf34f479b775878974f/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/OptimizationActivity.java

### PR DESCRIPTION
**Pull request by langsmith**
This pr resolves #358 by adding a new page and written docs for [the Optimization API](https://www.mapbox.com/api-documentation/#optimization). The page largely matches the Geocoding page.

screen shot 2018-03-16 at 5 01 50 pm
screen shot 2018-03-16 at 5 01 52 pm

### CODE REVIEW

File: src/pages/mapbox-services/examples/index.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ description: 'Mapbox Java Services code examples'
 sideNavSections:
   - title: "Routing"
   - title: "Geocoding"
+  - title: "Optimization"  
   - title: "Static Maps"
 ---*/
```
Comment (cammace) [inline src/pages/mapbox-services/examples/index.js]: Any reason for this being commented out?

Comment (langsmith) [reply src/pages/mapbox-services/examples/index.js]: I think it's a Batfish thing. The Maps SDK and Plugin example pages have that top part commented out as well. Uncommenting and then testing site locally throws the follow error in `Terminal`. Seems that we need to keep it commented out for now:

screen shot 2018-03-19 at 9 35 29 am

Comment (colleenmcginnis) [reply src/pages/mapbox-services/examples/index.js]: Yes this is a batfish thing. For `.js` files the front matter should be commented out. For `.md` files they should not be commented out. See the [batfish documentation](https://github.com/mapbox/batfish/#js-pages).

File: src/pages/mapbox-services/overview/optimization.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+1. **A valid Mapbox [access token](https://www.mapbox.com/help/define-access-token/)**.
+2. **A `<List>` of `Point` objects**. These are the waypoints in the optimized route. The minimum amount of `Point` objects in the list is 2, and the maximum is 12.
+3. **A directions profile**. The API needs to know whether you want a route which is optimized for driving, cycling, or walking.
+
+Here's an example `MapboxOptimization` object:
+
+```java
+MapboxOptimization optimizedClient = MapboxOptimization.builder()
+  .coordinates(coordinates)
```
Comment (colleenmcginnis) [inline src/pages/mapbox-services/overview/optimization.md]: Would it be helpful to show an example with coordinates so users know how to format them?

Comment (langsmith) [reply src/pages/mapbox-services/overview/optimization.md]: The coordinates can be built in many different ways. The demo app has a more fleshed out optimization API example anyways. I think a good compromise here in this page is the commit I just made which adds a little preview above the `MapboxOptimization` builder to show that `coordinates` is `List<Point>`. 👍 ? 👎 ?

```java
private List<Point> coordinates;
...
MapboxOptimization optimizedClient = MapboxOptimization.builder()
  .coordinates(coordinates)
  .profile(DirectionsCriteria.PROFILE_DRIVING)
  .accessToken(Mapbox.getAccessToken())
  .build();
```

**Comment by langsmith** [comment]
All set @colleenmcginnis

**Comment by langsmith** [comment]
Currently blocked by work mentioned in https://github.com/mapbox/android-docs/pull/375#discussion_r176138002

**Comment by langsmith** [comment]
I made some final small tweaks @colleenmcginnis . Please 👀 if you would and then we can merge.

**Pull request merged**